packages/sdk-js/src/GrowthBook.ts

```diff
@@ -1149,4 +1297,151 @@ export class GrowthBook<
       undo.forEach((fn) => fn());
     };
   }
+
+  private _deriveStickyBucketIdentifierAttributes(data?: FeatureApiResponse) {
+    const attributes = new Set<string>();
+    const features = data
+      ? data.features ?? this.getFeatures()
+      : this.getFeatures();
+    const experiments = data
+      ? data.experiments ?? this.getExperiments()
+      : this.getExperiments();
+    Object.keys(features).forEach((id) => {
+      const feature = features[id];
+      if (feature.rules) {
+        for (const rule of feature.rules) {
+          if (rule.variations) {
+            attributes.add(rule.hashAttribute || "id");
+            if (rule.fallbackAttribute) {
+              attributes.add(rule.fallbackAttribute);
+            }
+          }
+        }
+      }
+    });
+    experiments.map((experiment) => {
+      attributes.add(experiment.hashAttribute || "id");
+      if (experiment.fallbackAttribute) {
+        attributes.add(experiment.fallbackAttribute);
+      }
+    });
+    return Array.from(attributes);
+  }
+
+  public async refreshStickyBuckets(data?: FeatureApiResponse) {
+    if (this.context.stickyBucketService) {
+      const attributes = this._getStickyBucketAttributes(data);
+      this.context.stickyBucketAssignmentDocs = await this.context.stickyBucketService.getAllAssignments(
+        attributes
+      );
+    }
+  }
+
+  private _getStickyBucketAssignments(): StickyAssignments {
+    const mergedAssignments: StickyAssignments = {};
+    Object.entries(this.context.stickyBucketAssignmentDocs ?? {}).forEach(
+      ([, doc]) => {
+        if (doc.assignments) Object.assign(mergedAssignments, doc.assignments);
+      }
+    );
+    return mergedAssignments;
+  }
+
+  private _getStickyBucketVariation(
+    experimentKey: string,
+    experimentBucketVersion: number = 1,
+    minExperimentBucketVersion: number = 0,
+    meta: VariationMeta[] = []
```

Comment:
Minor code golf: default attribute values tend to be very verbose when transpiled